### PR TITLE
Show previous visibility specifier in parser error

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/libsolidity/SolidityParser.cpp [hunk 1]
```diff
@@ -898,7 +898,12 @@ BOOST_AUTO_TEST_CASE(multiple_visibility_specifiers)
 		contract c {
 			uint private internal a;
 		})";
-	CHECK_PARSE_ERROR(text, "Visibility already specified");
+	CHECK_PARSE_ERROR(text, "Visibility already specified. Previous: \"private\".");
```
Comment (axic) [inline test/libsolidity/SolidityParser.cpp]: Alternative: `"Visibility already specified as "private".`

File: libsolidity/parsing/Parser.cpp [hunk 2]
```diff
@@ -307,6 +307,24 @@ Declaration::Visibility Parser::parseVisibilitySpecifier(Token::Value _token)
 	return visibility;
 }
 
+string Parser::visibilitySpecifierName(Declaration::Visibility _visibility)
```
Comment (chriseth) [inline libsolidity/parsing/Parser.cpp]: What about making that a function `toString`, declared closer to `Declaration::Visibility`? I think we use that also in other places.

File: test/libsolidity/SolidityParser.cpp (hunk 1 above)
Comment (chriseth) [reply test/libsolidity/SolidityParser.cpp]: I prefer the alternative.

File: libsolidity/parsing/Parser.cpp (hunk 2 above)
Comment (axic) [reply libsolidity/parsing/Parser.cpp]: Do declared in `AST.h` on the global level?

Comment (chriseth) [reply libsolidity/parsing/Parser.cpp]: or a static function of Declaration

Comment (chriseth) [reply libsolidity/parsing/Parser.cpp]: astjsonconverter at least has a copy of it.

Comment (axic) [reply libsolidity/parsing/Parser.cpp]: See the last commit, though changing that now to a static function.

**Review by chriseth — APPROVED** [review]

**Pull request merged**